ast-nodes/expressions/operators/less-operator.dart

```diff
@@ -30,6 +34,21 @@ class LessOperator extends BinaryRelation implements Comparison {
   }
 
   void checkSemantics() {
-    // TODO: implement
+    leftOperand.checkSemantics();
+    rightOperand.checkSemantics();
+
+    if (leftOperand.resultType is RealType ||
+        rightOperand.resultType is RealType) {
+      this.leftOperand = ensureType(this.leftOperand, RealType());
+      this.rightOperand = ensureType(this.rightOperand, RealType());
+    } else if (leftOperand.resultType is IntegerType ||
+        rightOperand.resultType is IntegerType) {
+      this.leftOperand = ensureType(this.leftOperand, IntegerType());
+      this.rightOperand = ensureType(this.rightOperand, IntegerType());
+    } else {
+      if (leftOperand.resultType != rightOperand.resultType) {
+        throw SemanticError(this, "Cannot compare objects of different types");
```

Comment:
We cannot compare arrays, for example, even if they are of the same type. I suppose we can compare booleans, but that's up to you (a _decision_)

Reply:
So, what's the problem here?

Reply:
The problem is your checker will allow comparing `array [5] integer` and `array [5] integer`, but they are incomparable